File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.cpp
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
        unordered_map<string,int> mp{};
        int n=grid.size();
        int cnt{};
        
        for(auto i:grid){
            string s="";
            for(int& j:i)
                s+=(j+'0');
            ++mp[s];
        }
        
        for(int i=0;i<n;++i){
            string s="";
            for(int j=0;j<n;++j)
                s+=(grid[j][i]+'0');
            cnt+= mp[s];
        }
        return cnt;
    }
};

File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.java
class Solution {
    public int equalPairs(int[][] grid) {
        int cnt=0;
        int n=grid.length;
        
        for(int[] row:grid)
            for(int j=0, i=0;j<n;++j){
                for(i=0;i<n;++i)
                    if(grid[i][j]!=row[i])
                        break; 
                if(i==n) ++cnt;
            }
        return cnt;
    }
}